### ISSUE TITLE
Getting an existing service is missing important properties

### ISSUE THREAD

**Issue by manast** [pulumi/pulumi-kubernetes#1290]
For some reason, getting the istio service (installed as an addon in GKE) is missing important properties such as status and metadata:

```
const ingress = k8s.core.v1.Service.get(
      "ingress",
      "istio-system/istio-ingressgateway",
      { provider }
    );
```

```
 Service {
        __pulumiResource: true,
        __transformations: [],
        __name: 'ingress',
        __providers: { kubernetes: [Provider] },
        __protect: false,
        __aliases: [],
        urn: OutputImpl {
          __pulumiOutput: true,
          resources: [Function],
          allResources: [Function],
          isKnown: [Promise],
          isSecret: [Promise],
          promise: [Function],
          toString: [Function],
          toJSON: [Function]
        },
        id: OutputImpl {
          __pulumiOutput: true,
          resources: [Function],
          allResources: [Function],
          isKnown: [Promise],
          isSecret: [Promise],
          promise: [Function],
          toString: [Function],
          toJSON: [Function]
        },
        __pulumiCustomResource: true,
        __pulumiType: 'kubernetes:core/v1:Service'
      },
```

In my case I am interested in the metadata object, the reason is that I would like to add some annotations while preserving the old metadata:

```
    const metadata = ingress.metadata.apply((metadata) => ({
      ...metadata,
      annotations: {
        "external-dns.alpha.kubernetes.io/hostname": `*.${location}.${domain}.`,
      },
    }));

    const istioIngress = new k8s.core.v1.Service(
      "ingress",
      {
        metadata,
      },
      { provider, import: "istio-system/istio-ingressgateway" }
    );
```

**Comment by leezen** [comment]
Could you please describe where you're not seeing the metadata? Are you saying that `ingress.metadata` is an empty object?

**Comment by manast** [comment]
yes, the full ingress object is shown in the issue, no status and no metadata.

**Comment by marioapardo** [comment]
The same happens with the secrets that ServiceAccount generates when trying to obtain the token, the answer is none

**Comment by leezen** [comment]
@manast And you're able to obtain those properties via `kubectl`? The `get` call is using the k8s API, so I would be surprised if there's a difference in the underlying data.

**Comment by marioapardo** [comment]
@leezen With kubectl you get the data without problems. We currently use the python sdk.

**Comment by marioapardo** [comment]
This is the output of the complete resource that you get 
`cerberus_token  : {
        id : "kube-system/cerberus-token-q7hph"
        urn: "urn:pulumi:cerberus::sandbox-eks-pre::cerberus:eks:Cluster$cerberus:eks:Master$kubernetes:core/v1:Secret::sandbox-eks-pre-cerberus-aws-k8s-cerberus-secret"
    }`

**Comment by manast** [comment]
> @manast And you're able to obtain those properties via `kubectl`? The `get` call is using the k8s API, so I would be surprised if there's a difference in the underlying data.

kubectl is just working fine...

**Comment by iridian-ks** [comment]
I noticed the exact same behavior. Certain attributes on Secrets were simply missing. Was also using the Python SDK.

I did straight pulumi.debugs of the Secret objects and only the `id` attribute was available.

Upgrading Pulumi and the k8s provider to the latest resolved it for me. Looking at the pulumi.debugs all of the attributes are there. My guess is this must have been a bug in the Python SDK with data bindings.
